configure.ac

```diff
@@ -35,12 +35,30 @@ PRESET_LDFLAGS="$LDFLAGS"
 dnl Set default LDFLAGS
 AS_IF([test "x$LDFLAGS" = "x"],[LDFLAGS="-g"])
 
-# Check for GNU cc
+# check for host OS detail
+AC_CANONICAL_HOST
+AC_MSG_CHECKING([simplified host os])
+simple_host_os=`echo $host_os|sed 's/[0-9].*//g;s/-.*//g'`
+squid_host_os_version=`echo $host_os|tr -d "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-"`
+AS_IF([test -n "$squid_host_os_version"],[
+  squid_host_os="`echo $simple_host_os| sed s/$squid_host_os_version//g`"
+],[
+  squid_host_os="$simple_host_os"
+])
+AC_MSG_RESULT($squid_host_os (version $squid_host_os_version))
+# on windows squid_host_os is either mingw or cygwin, version is 32
+
 AC_PROG_CC
 AM_PROG_CC_C_O
 AC_PROG_CXX
+
+# this needs to be before any test is run, to have more standard
+# functions available on some Unix sysems (e.g. Solaris)
```

Comment:
The "any test" text is too general to be meaningful in this context -- clearly we have already performed some tests above. Let's use AC_USE_SYSTEM_EXTENSIONS documentation instead. I also do not recommend re-documenting What AC_USE_SYSTEM_EXTENSIONS does.

```suggestion
# AC_USE_SYSTEM_EXTENSIONS should be called before any macros that run the C compiler.
```

I realize that you are just moving an old comment, but we should take in to account that the old comment tried and _failed_ to prevent the bugs this PR is fixing. Improving it is in scope IMO, but I do not insist on this change.

P.S. I did write a very similar change request in the first review but it looks like I somehow deleted it back then. Sorry for one more iteration.